pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -274,13 +288,14 @@ func (rc *reconciler) reconcile() {
 			}
 		}
 	}
+}
 
-	// Ensure devices that should be detached/unmounted are detached/unmounted.
+func (rc *reconciler) unmountDetachDevices() {
 	for _, attachedVolume := range rc.actualStateOfWorld.GetUnmountedVolumes() {
 		// Check IsOperationPending to avoid marking a volume as detached if it's in the process of mounting.
 		if !rc.desiredStateOfWorld.VolumeExists(attachedVolume.VolumeName) &&
 			!rc.operationExecutor.IsOperationPending(attachedVolume.VolumeName, nestedpendingoperations.EmptyUniquePodName) {
```

Comment:
not your change, but I forgot why we need to check operationPending only for unmountDevice?

Reply:
Here is the PR that implemented this - https://github.com/kubernetes/kubernetes/pull/40417